stdlib/src/collections/list.mojo

```diff
@@ -610,3 +600,9 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
                 result += ", "
         result += "]"
         return result
+
+    @always_inline
+    fn _normalize_index(self, index: Int) -> Int:
+        if index < 0:
+            return _max(0, len(self) + index)
```

Comment:
Why do we need the `_max` here?

Reply:
It would be here just to prevent the negative indexes. How would you suggest to do it?

Reply:
Given that index outside `[-len(self), len(self))` is out of bound, we can simply skip the `_max` call.
```mojo
return index if index >= 0 else index + len(self)
```

Reply:
@JoeLoser @gabrieldemarmiesse can I get your views on this ?

Reply:
Hey @StandinKP — sorry for the slow reply here.  We discussed this the other day during our team meeting and here's my notes:

- The current code seems wrong because it will silently turn an out of bounds negative value into an index of the 0th element by using `_max`.  But  I don’t think the solution is to remove the `_max` and allow negative indices to segfault (or worse) at runtime.  Instead, let's `abort` on out-of-bound indices. 
- In the future, it would be nice to have one "normalized index" function for use in the library for any contiguous data structure we have in the library such as `List`, `String`, etc.  Nothing to do for this PR, but just pointing out how we'd like to clean this up over time.
- In the short (and long-term), I think aborting on out-of-bounds access is a reasonable approach, both here in `List` and elsewhere.

Reply:
Sure. Makes sense. Do you want me to keep an issue filed to add the common "normalized function" and update all references?